Error building project: 'Range' is an ambiguous reference

"make" returns this error for 6 source files:
 > error CS0104: 'Range' is an ambiguous reference between 'System.Range' and 'Bless.Util.Range'

I have solved the issue editing the sources and explicitly choosing which I guess is the right class:
> using Range=Bless.Util.Range;

The files where the compiler complains are:
> src/buffers/ByteBuffer.cs
> src/gui/DataBookFinder.cs
> src/gui/areas/AreaGroup.cs
> src/tools/find/FindOperations.cs
> src/tools/find/IFindStrategy.cs
> src/tools/find/IFinder.cs

My mono version is:
> Mono JIT compiler version 6.4.0.198 (tarball Tue Sep 24 01:20:30 UTC 2019)

--- a/src/buffers/ByteBuffer.cs
+++ b/src/buffers/ByteBuffer.cs
@@ -671,7 +671,7 @@ public class ByteBuffer : BaseBuffer {
 	/// Returns as a SegmentCollection the data contained in 
 	/// the specified range in the buffer.  
 	///</summary>
-	public SegmentCollection RangeToSegmentCollection(Range range)
+	public SegmentCollection RangeToSegmentCollection(Util.Range range)
 	{
 		if (range.Size == 0)
 			return null;

--- a/src/gui/DataBookFinder.cs
+++ b/src/gui/DataBookFinder.cs
@@ -234,7 +234,7 @@ public class DataBookFinder : IFinder
 	{
 		GenericFindOperation state = (GenericFindOperation)ar.AsyncState;
 		ThreadedAsyncOperation.OperationResult result = state.Result;
-		Range match = state.Match;
+		Util.Range match = state.Match;
 
 		DataView dv = null;
 
@@ -381,7 +381,7 @@ public class DataBookFinder : IFinder
 	{
 		ReplaceAllOperation state = (ReplaceAllOperation)ar.AsyncState;
 		ThreadedAsyncOperation.OperationResult result = state.Result;
-		Range firstMatch = state.FirstMatch;
+		Util.Range firstMatch = state.FirstMatch;
 
 		DataView dv = null;
 

--- a/src/gui/areas/AreaGroup.cs
+++ b/src/gui/areas/AreaGroup.cs
@@ -198,7 +198,7 @@ public class AreaGroup
 		set { drawingArea = value; }
 	}
 	
-	public Range Selection {
+	public Util.Range Selection {
 		get { return selection; }
 		set { 
 			if (selection == value)
@@ -251,7 +251,7 @@ public class AreaGroup
 	/// Get the range of bytes and the number of rows that 
 	/// are displayed in the current view.
 	/// </summary>
-	public Range GetViewRange(out int nrows)
+	public Util.Range GetViewRange(out int nrows)
 	{
 		// find out number of rows, bytes in current view
 		
@@ -271,9 +271,9 @@ public class AreaGroup
 		
 		// make sure we get an empty clipping Range when bleft==0
 		if (bleft > 0)
-			return new Range(offset, offset + bleft - 1);
+			return new Util.Range(offset, offset + bleft - 1);
 		else
-			return new Range();
+			return new Util.Range();
 	}
 	
 	/// <summary>
@@ -381,7 +381,7 @@ public class AreaGroup
 	private void SetupBufferCache()
 	{
 		int nrows;
-		Range view = GetViewRange(out nrows);
+		Util.Range view = GetViewRange(out nrows);
 		if (view.Size != bufferCache.Length)
 			bufferCache = new byte[view.Size];
 		
@@ -444,7 +444,7 @@ public class AreaGroup
 		}
 	}
 	/// <summary>
-	/// Renders a <see cref="Range"/> of data using a specified <see cref="Drawer.HighlightType"/>
+	/// Renders a <see cref="Util.Range"/> of data using a specified <see cref="Drawer.HighlightType"/>
 	/// </summary>
 	private void RenderHighlight(AtomicHighlight h)
 	{
@@ -474,12 +474,12 @@ public class AreaGroup
 		
 		if (q.Type > r.Type) {
 			ha = new AtomicHighlight[3]{new AtomicHighlight(r), new AtomicHighlight(q), new AtomicHighlight(r)};
-			Range.SplitAtomic(ha, r, q);
+			Util.Range.SplitAtomic(ha, r, q);
 			ha[1].AddContainer(r);
 		}
 		else {
 			ha = new AtomicHighlight[3]{new AtomicHighlight(q), new AtomicHighlight(r), new AtomicHighlight(q)};
-			Range.SplitAtomic(ha, q, r);
+			Util.Range.SplitAtomic(ha, q, r);
 			foreach (Highlight h in q.Containers)
 				ha[1].AddContainer(h);
 		}
@@ -533,7 +533,7 @@ public class AreaGroup
 	private IntervalTree<AtomicHighlight> GetAtomicHighlights()
 	{
 		int nrows;
-		Range clip = GetViewRange(out nrows);
+		Util.Range clip = GetViewRange(out nrows);
 		Highlight view = new Highlight(clip, Drawer.HighlightType.Normal);
 		
 		// get all highlights in current view
@@ -618,7 +618,7 @@ public class AreaGroup
 		// if we don't find an overlap this means that either
 		// 1. the prev cursor position is not visible on the screen
 		// 2. the prev cursor position is at or beyond the end of the file
-		IList<AtomicHighlight> overlaps = atomicHighlights.SearchOverlap(new Range(prevCursorOffset, prevCursorOffset));
+		IList<AtomicHighlight> overlaps = atomicHighlights.SearchOverlap(new Util.Range(prevCursorOffset, prevCursorOffset));
 		
 		AtomicHighlight h = null;
 		

--- a/src/tools/find/FindOperations.cs
+++ b/src/tools/find/FindOperations.cs
@@ -146,8 +146,8 @@ public class ReplaceAllOperation: GenericFindOperation
 
 	protected override void DoOperation()
 	{
-		Range m;
-		match = new Range();
+		Util.Range m;
+		match = new Util.Range();
 		firstMatch = null;
 
 		numReplaced = 0;
@@ -159,7 +159,7 @@ public class ReplaceAllOperation: GenericFindOperation
 
 		while ((m = strategy.FindNext()) != null) {
 			if (firstMatch == null) {
-				firstMatch = new Range(m);
+				firstMatch = new Util.Range(m);
 			}
 
 			match.Start = m.Start;

--- a/src/tools/find/IFindStrategy.cs
+++ b/src/tools/find/IFindStrategy.cs
@@ -31,11 +31,11 @@ public interface IFindStrategy
 			ByteBuffer Buffer {	get; set; }
 				long Position { get; set; }
 
-						Range FindNext();
-	Range FindPrevious();
+						Util.Range FindNext();
+	Util.Range FindPrevious();
 
-	Range FindNext(long limit);
-	Range FindPrevious(long limit);
+	Util.Range FindNext(long limit);
+	Util.Range FindPrevious(long limit);
 
 	bool Cancelled { get; set; }
 		}

--- a/src/tools/find/IFinder.cs
+++ b/src/tools/find/IFinder.cs
@@ -30,7 +30,7 @@ public delegate void FirstFindHandler();
 public interface IFinder
 {
 	IFindStrategy Strategy { get; set; }
-			Range LastFound { get; set; }
+			Util.Range LastFound { get; set; }
 
 
 					event FirstFindHandler FirstFind;

